src/test/java/de/tum/in/www1/artemis/AssessmentComplaintIntegrationTest.java

```diff
@@ -737,4 +773,24 @@ public void testGetComplaintsByCourseIdAndExamId() throws Exception {
         assertThat(fetchedComplaints.get(0).getId()).isEqualTo(storedComplaint.get().getId().intValue());
         assertThat(fetchedComplaints.get(0).getComplaintText()).isEqualTo(storedComplaint.get().getComplaintText());
     }
+
+    @Test
+    @WithMockUser(username = "student1", roles = "USER")
+    public void submitComplaintForExerciseComplaintExceededTextLimit() throws Exception {
+        course = database.updateCourseComplaintTextLimit(course, 25);
+        // 26 characters
+        complaint.setComplaintText("abcdefghijklmnopqrstuvwxyz");
+        request.post("/api/complaints", complaint, HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    @WithMockUser(username = "student1", roles = "USER")
+    public void submitComplaintForExerciseComplaintNotExceededTextLimit() throws Exception {
+        course = database.updateCourseComplaintTextLimit(course, 27);
+        // 26 characters
+        complaint.setComplaintText("abcdefghijklmnopqrstuvwxyz");
+        request.post("/api/complaints", complaint, HttpStatus.CREATED);
+        Optional<Complaint> storedComplaint = complaintRepo.findByResultId(modelingAssessment.getId());
+        assertThat(storedComplaint).as("complaint is saved").isPresent();
```

Comment:
Question: What does this actually do?
The assertThat().as("...") looks weird to me.

Reply:
I agree. There is no reason for that. I just oriented by previous implementations. But I think it's best to remove it here.